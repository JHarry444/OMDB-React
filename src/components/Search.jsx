import axios from 'axios';
import { Alert } from 'react-bootstrap';
import { useHistory } from 'react-router-dom';
import { useState } from 'react';
import SearchResults from './SearchResults';
import SearchBar from './SearchBar';

const Search = () => {
  const { push } = useHistory();

  const [searchTerm, setSearchTerm] = useState('');
  const [films, setFilms] = useState([]);
  const [error, setError] = useState('');

  const handleSearch = (event) => {
    event.preventDefault();
    axios.get(`http://www.omdbapi.com/?apikey=335035be&s=${searchTerm}`)
      .then((response) => {
        if (response.data.Error) {
          setError(response.data.Error);
        } else {
          setFilms(response.data.Search);
          setError('');
        }
      }).catch((err) => {
        setError(err);
      });
  };

  const getDetails = (id) => {
    push(`/details/${id}`);
  };

  return (
    <div>
      <SearchBar
        searchTerm={searchTerm}
        handleChange={(e) => setSearchTerm(e.target.value)}
        searchFunc={handleSearch}
      />
      {error
        ? (
          <Alert variant="danger" onClose={() => setError('')} dismissible>
            <Alert.Heading>{error}</Alert.Heading>
          </Alert>
        )
        : <SearchResults films={films} getDetails={getDetails} />}
    </div>
  );
};

export default Search;
