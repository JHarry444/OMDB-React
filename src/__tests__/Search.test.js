import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Search from '../Search';


it('renders without crashing', () => {
  shallow(<Search/>);
});

it('matches snapshot', () => {
  const tree = renderer.create(<Search />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('has query', () => {
    const search = shallow(<Search/>);
    expect
})

