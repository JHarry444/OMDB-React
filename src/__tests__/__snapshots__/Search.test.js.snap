// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="input-group"
    >
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Enter search term here"
      />
      <div
        className="input-group-append"
      >
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Button
        </button>
      </div>
    </div>
  </form>
  <div
    className="container"
  >
    <div
      className="row"
    />
  </div>
</div>
`;
